File: media/forms/EXM100/frmEXM100.js
define([
    "dojo/_base/declare",
    "common/MinimalForm",
    "dojo/text!./frmEXM100.html",
    "dojo/on",
], function(declare, MinimalForm, template, on) {

    return declare([MinimalForm], {
        metag: 'EXM100',
        baseURL: 'EXM100',
        templateString: template,

        startup: function() {
            this.inherited(arguments);
            var widget = this;
            on(widget.cmdTest1, 'click', function(){
                // var int1 = widget.txtint1.get('value')
                // var int2 = widget.txtint2.get('value')
                var data = {};
                widget.param(data,'txtint1',['value']);
                widget.param(data,'txtint2',['value']);
                widget.post('LCM',data,function(result){
                    
                widget.txtresult.set('value',result.lcm)
                });
            });

            on(widget.cmdTest2, 'click', function(){
                // var int1 = widget.txtint1.get('value')
                // var int2 = widget.txtint2.get('value')
                var data = {};
                widget.param(data,'txtint1',['value']);
                widget.param(data,'txtint2',['value']);
                widget.post('GCD',data,function(result){
                    
                    widget.txtresult.set('value',result.gcd)
                });
            });

            on(widget.cmdclr, 'click', function(){
                // var int1 = widget.txtint1.get('value')
                // var int2 = widget.txtint2.get('value')
              
                widget.txtint1.set('value',"")
                widget.txtint2.set('value',"")
                widget.txtresult.set('value',"")
            });

            on(widget.cmdclk, 'click', function(){
                // var int1 = widget.txtint1.get('value')
                // var int2 = widget.txtint2.get('value')
                var data = {};
                widget.param(data,'txt1',['value']);
                widget.param(data,'txt2',['value']);
                widget.post('WORDCOU',data,function(result){
                    
                    widget.txtresult2.set('value',result.word)
                });
            });

        },
    });
});

File: media/forms/EXM100/frmEXM100.py
from django.shortcuts import render, HttpResponse
import json

def load(request):
    return render(request,'minimal-frame.html',{'widget':'forms/EXM100/frmEXM100'})

def LCM(request):
    params = json.loads(request.body)
    num1 = int(params['txtint1']['value'])
    num2 = int(params['txtint2']['value'])

    lc = num1 * num2 / gcd(num1,num2)
    result = {'lcm':lc}
    return HttpResponse(json.dumps(result))

def GCD(request):
    params = json.loads(request.body)
    num1 = int(params['txtint1']['value'])
    num2 = int(params['txtint2']['value'])

    gc = gcd(num1,num2)
    result = {'gcd':gc}
    return HttpResponse(json.dumps(result))

def gcd(a,b):
    while b > 0:
        a,b = b,a % b
    return a

def WORDCOU(request):
    params = json.loads(request.body)
    txt11 = params['txt1']['value']
    txt22 = params['txt2']['value']

    
    count1 = txt11.count(txt22)

    result = {'word':count1}
    return HttpResponse(json.dumps(result))


